[#593] Add HTTP ILB module (net-ilb-l7)

LucaPrete opened the pull request.

ludoo commented on modules/net-ilb-l7/README.md:
````diff
@@ -0,0 +1,398 @@
+# tftest modules=1 resources=5
+```
+
+Otherwise, unamanged certificates can also be contextually created:
````
typo

LucaPrete replied:
Done

ludoo commented on modules/net-ilb-l7/README.md:
```diff
@@ -0,0 +1,398 @@
+      domains = [
+        "my-domain.com"
+      ],
+      tls_private_key      = nonsensitive(tls_private_key.self_signed_key.private_key_pem)
```
do we really need nonsensitive on the key?

LucaPrete replied:
I'd leave it as it is. It's consistent with what we already do for the glb and it reflects what's prescribed in the providers readme.

LucaPrete commented on modules/net-ilb-l7/README.md:
```diff
@@ -0,0 +1,398 @@
+
+  backend_services_config = {
+    my-backend-svc = {
+      enable_cdn    = false
```
This was present in the regional provider, so initially I left it there...anyway it makes completely sense to remove it. Done!

LucaPrete commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@ All notable changes to this project will be documented in this file.
 
 ## Unreleased
 
+- new [net-ilb-l7](modules/net-ilb-l7) module
```
Done

LucaPrete commented on modules/net-ilb-l7/README.md:
````diff
@@ -0,0 +1,398 @@
+# tftest modules=1 resources=5
+```
+
+### Network Endpoint Groups and Serverless Backends
````
Commented the code in the readme and left a comment instead. We'll reactivate this, as soon as it will be possible.

ludoo replied:
NEGs are supported, it's serveless which isn't AFAIK. Can we add support for NEGs, or document it if it's already in?

LucaPrete replied:
You're right! Put back (and tested) support for zonal NEGs.

sruffilli commented:
Got the following errors when trying to apply using the currently provided example code:
```
│ Error: Error creating RegionHealthCheck: googleapi: Error 400: Invalid value for field 'resource.name': 'ilb-l7-hc_1'. Must be a match of regex '(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?)', invalid
```

```
│ Error: Error creating RegionHealthCheck: googleapi: Error 400: Invalid value for field 'resource.timeoutSec': '40'. TimeoutSec should be less than checkIntervalSec., invalid
```

LucaPrete commented:
@sruffilli thanks for testing.

* Fixed the examples runtime errors
* Extracted network and subnetwork from forwarding rule variables (now at root level for simplicity)
* Added and tested shared vpc example

ludoo commented on modules/net-ilb-l7/outputs.tf:
```diff
@@ -0,0 +1,55 @@
+  value       = try(google_compute_region_ssl_certificate.certificates, null)
+}
+
+output "ip_address" {
```
This should probably have the forwarding rule IP, either reserved or dynamically generated.
